src/purchases.ts

```diff
@@ -905,6 +906,117 @@ export default class Purchases {
     );
   }
 
+  /**
+   * iOS only. Use this function to retrieve the eligible `PurchasesWinBackOffer`s that a subscriber
+   * is eligible for for a given `PurchasesProduct`.
+   *
+   * @param product The `PurchasesStoreProduct` the user intends to purchase.
+   * @returns { Promise<[PurchasesWinBackOffer]> } Returns an array of win-back offers that the subscriber is eligible for.
+   * Null is returned for Android and incompatible iOS versions. The promise will be rejected if configure has not been
+   * called yet or if there's an error getting the payment discount.
+   */
+  public static async getEligibleWinBackOffersForProduct(
+    product: PurchasesStoreProduct
+  ): Promise<[PurchasesWinBackOffer] | undefined> {
+    await Purchases.throwIfNotConfigured();
+    if (Platform.OS === "android") {
+      return Promise.resolve(undefined);
+    }
+
+    return RNPurchases.eligibleWinBackOffersForProductIdentifier(
+      product.identifier
+    );
+  }
+
+  /**
+   * iOS only. Use this function to retrieve the eligible `PurchasesWinBackOffer`s that a subscriber
+   * is eligible for for a given `PurchasesPackage`.
+   *
+   * @param aPackage The `PurchasesPackage` the user intends to purchase.
+   * @returns { Promise<[PurchasesWinBackOffer]> } Returns an array of win-back offers that the subscriber is eligible for.
+   * Null is returned for Android and incompatible iOS versions. The promise will be rejected if configure has not been
+   * called yet or if there's an error getting the payment discount.
+   */
+  public static async getEligibleWinBackOffersForPackage(
+    aPackage: PurchasesPackage
+  ): Promise<[PurchasesWinBackOffer] | undefined> {
+    await Purchases.throwIfNotConfigured();
+    if (Platform.OS === "android") {
+      return Promise.resolve(undefined);
+    }
+
+    return RNPurchases.eligibleWinBackOffersForProductIdentifier(
+      aPackage.product.identifier
```

Comment:
Here, we call the `eligibleWinBackOffersForProductIdentifier` for the package's product ID. We do this so that we only have to implement the offer caching logic in PHC once instead of twice for both functions